### PR TITLE
[202311] Use shared_ptr for DbConnector objects in PortsOrch and FdbOrch

### PR DESCRIPTION
**Pull request by saiarcot895**
**What I did**

In the constructor of PortsOrch and FdbOrch, a pointer to a DbConnector object is created, then used in a notifications consumer. However, because the ownership technically belongs to PortsOrch and FdbOrch, and these are local variables, nothing frees the DbConnector object if instances of these classes get destroyed. This may result in a memory leak.

**Why I did it**

To fix this, wrap these objects in shared_ptr, and then make them instance variables. This makes sure that they get freed when instances of these classes get destroyed.

**How I verified it**

Loaded the master branch build in master branch image on KVM; no apparent errors, and BGP is up.

**Details if related**

### CODE REVIEW

File: orchagent/fdborch.cpp [hunk 1]
```diff
@@ -42,8 +42,8 @@ FdbOrch::FdbOrch(DBConnector* applDbConnector, vector<table_name_with_pri_t> app
     Orch::addExecutor(flushNotifier);
 
     /* Add FDB notifications support from ASIC */
-    DBConnector *notificationsDb = new DBConnector("ASIC_DB", 0);
-    m_fdbNotificationConsumer = new swss::NotificationConsumer(notificationsDb, "NOTIFICATIONS");
+    m_notificationsDb = shared_ptr<DBConnector>(new DBConnector("ASIC_DB", 0));
```
Comment (qiluo-msft) [inline orchagent/fdborch.cpp:45]: >shared_ptr [](http://example.com/codeflow?start=24&length=10)

`make_shared` can simplify this line of code.

File: orchagent/portsorch.cpp [hunk 2]
```diff
@@ -628,14 +628,14 @@ PortsOrch::PortsOrch(DBConnector *db, DBConnector *stateDb, vector<table_name_wi
     }
 
     /* Add port oper status notification support */
-    DBConnector *notificationsDb = new DBConnector("ASIC_DB", 0);
-    m_portStatusNotificationConsumer = new swss::NotificationConsumer(notificationsDb, "NOTIFICATIONS");
+    m_notificationsDb = shared_ptr<DBConnector>(new DBConnector("ASIC_DB", 0));
```
Comment (qiluo-msft) [inline orchagent/portsorch.cpp:631]: >shared_ptr [](http://example.com/codeflow?start=24&length=10)

`make_shared`

**Comment by qiluo-msft** [comment]
This is a valid bug fix on master and old branches. Let's fix master first, then backport to all old branches.

File: orchagent/fdborch.cpp [hunk 3]
```diff
@@ -42,8 +42,8 @@ FdbOrch::FdbOrch(DBConnector* applDbConnector, vector<table_name_with_pri_t> app
...
+    m_fdbNotificationConsumer = new swss::NotificationConsumer(m_notificationsDb.get(), "NOTIFICATIONS");
     auto fdbNotifier = new Notifier(m_fdbNotificationConsumer, this, "FDB_NOTIFICATIONS");
```
Comment (kcudnik) [inline orchagent/fdborch.cpp:47]: use make shared

**Comment by saiarcot895** [comment]
Closing in favor of #3017 being backported.

**Pull request closed without merging**